Introduced background and handbook organisation, reworked text a bit (#33)
neicnordic/neic-sda: docs/index.md

NeIC Sensitive Data Archive
===========================

> The NeIC Sensitive Data Archive (SDA) software is a modular system of microservices that in alternative configurations can be setup to host sensitive data in an archival service, securely stored encrypted at rest.

Reviewer: To me, this sentence is a bit complex for a first introduction to the project. Maybe something like:
```suggestion
The NeIC Sensitive Data Archive (SDA) is an encrypted data archive, originally implemented for storage of sensitive biological data. It is implemented as a modular microservice system that can be deployed in different configurations depending on the service needs.
```
would be easier to read for new users?

> Local EGA (also known as Federated EGA) instance as `LEGA`, or
> `LocalEGA`. In the context of NeIC we will refer to the LocalEGA as the
> `Sensitive Data Archive` or `SDA`.

Overall architecture
--------------------

The main components and interaction partners of the NeIC Sensitive Data Archive deployment in a Federated EGA setup, are illustrated in the figure below. The different colored backgrounds represent different zones of separation in the federated deployment. 

![](https://docs.google.com/drawings/d/e/2PACX-1vSCqC49WJkBduQ5AJ1VdwFq-FJDDcMRVLaWQmvRBLy7YihKQImTi41WyeNruMyH1DdFqevQ9cgKtXEg/pub?w=1440&amp;h=810)

The components illustrated can be classified by which archive sub-process they take part in:

-   Submission - the process of submitting sensitive data and meta-data to the inbox staging area
-   Ingestion - the process of verifying uploaded data and securely storing it in archive storage, while synchronizing state and identifier information with CEGA
-   Data Retrieval - the process of re-encrypting and staging data for retrieval/download.

Service/component | Description | Archive sub-process 
-------:|:------------|:-----------------------------

> db | A Postgres database with appropriate schema, stores the file header the accession id, file path and checksums as well as other relevant information. | Submission, Ingestion and Data Retrieval 

Reviewer: ```suggestion
db | A Postgres database with appropriate schema, stores the file header, the accession id, file path and checksums as well as other relevant information. | Submission, Ingestion and Data Retrieval 
```

mq (broker) | A RabbitMQ message broker with appropriate accounts, exchanges, queues and bindings. We use a federated queue to get messages from CentralEGA's broker and shovels to send answers back.| Submission and Ingestion 
Inbox | Upload service for incoming data, acting as a dropbox. Uses credentials from Central EGA. | Submission 

> Intercept | relays message between the queue provided from the federated service and local queues. | Submission and Ingestion 

Reviewer: ```suggestion
Intercept | Relays messages between the queue provided from the federated service and local queues. | Submission and Ingestion 
```

> [Ingest](services/ingest.md) | Splits the Crypt4GH header and moves it to database. The remainder of the file is sent to the storage backend (archive). No cryptographic tasks are done. | Ingestion 

Reviewer: ```suggestion
[Ingest](services/ingest.md) | Splits the Crypt4GH header and moves it to the database. The remainder of the file is sent to the storage backend (archive). No cryptographic tasks are done. | Ingestion 
```

> [Verify](services/verify.md) | Uses a crypt4gh secret key, this service can decrypt the stored files and checksum them against the embedded checksum for the unencrypted file. | Ingestion 

Reviewer: ```suggestion
[Verify](services/verify.md) | Using the archive crypt4gh secret key, this service can decrypt the stored files and checksum them against the embedded checksum for the unencrypted file. | Ingestion 
```
